### PR TITLE
Build benchmarks in RMM CI

### LINKED ISSUE #940: [BUG] Benchmarks are not being built in CI
**Describe the bug**
The BUILD_BENCHMARKS cmake option is not being set when building RMM in CI.

**Expected behavior**
This option should be enabled.

**Environment details (please complete the following information):**
CI

**Additional context**
Discovered because after CUB / Thrust were updated to 1.15, RMM's build fails if BUILD_BENCHMARKS is enabled.

### PR DESCRIPTION
**Pull request by harrism**
Fixes #940

Adds options to build tests and benchmarks to build.sh, and uses them in ci/gpu/build.sh

### CODE REVIEW

**Review by Ethyling — APPROVED** [review]

File: build.sh [hunk 1]
```diff
@@ -154,7 +162,9 @@ fi
 if (( NUMARGS == 0 )) || hasArg librmm; then
     ensureCMakeRan
     echo "building librmm..."
-    cmake --build "${LIBRMM_BUILD_DIR}" -j${PARALLEL_LEVEL} ${VERBOSE_FLAG}
+    cmake --build "${LIBRMM_BUILD_DIR}" -j${PARALLEL_LEVEL} ${VERBOSE_FLAG} \
+        -DBUILD_TESTS=${BUILD_TESTS} \
+        -DBUILD_BENCHMARKS=${BUILD_BENCHMARKS}
```
Comment (Ethyling) [inline build.sh:167]: This should be move to the `ensureCMakeRan` function

**Review by Ethyling — CHANGES_REQUESTED** [review]

File: build.sh (hunk 1 above)
Comment (harrism) [reply build.sh:167]: Why wasn't it there already? All I did was add arguments.

Comment (Ethyling) [reply build.sh:167]: Sorry, I mean the new arguments should be in `ensureCMakeRan` with the other arguments. This line is running the build. The CMake configuration is done in `ensureCMakeRan`

File: build.sh [hunk 2]
```diff
@@ -42,6 +42,8 @@ BUILD_DIRS="${LIBRMM_BUILD_DIR} ${RMM_BUILD_DIR}"
 VERBOSE_FLAG=""
 BUILD_TYPE=Release
 INSTALL_TARGET=install
+BUILD_BENCHMARKS=OFF
```
Comment (robertmaynard) [inline build.sh:45]: We should add these new arguments to the help documentation

Comment (harrism) [reply build.sh:47]: Done.

File: build.sh (hunk 1 above)
Comment (harrism) [reply build.sh]: Done.

**Comment by Ethyling** [comment]
Following what is done in other repos, you might need to update the conda build scripts to also build tests if Project Flash is enabled. See https://github.com/rapidsai/cudf/blob/branch-22.02/conda/recipes/libcudf/build.sh as example

**Comment by harrism** [comment]
@Ethyling that seems out of scope for this PR. This PR is about building benchmarks, not tests. Also I'm out sick and we are in burndown. so could you approve the present PR and just file an issue for this detail?

**Comment by Ethyling** [comment]
> @Ethyling that seems out of scope for this PR. This PR is about building benchmarks, not tests. Also I'm out sick and we are in burndown. so could you approve the present PR and just file an issue for this detail?

I think this is required to fix this CI issue: `ci/gpu/build.sh: line 106: build/gtests/*: No such file or directory`

Before this PR, the default value of `BUILD_TESTS` in CMake was `ON`, see: https://github.com/rapidsai/rmm/blob/branch-22.02/CMakeLists.txt#L38
Now it's `OFF` due to these changes:
- https://github.com/rapidsai/rmm/pull/941/files#diff-4d2a8eefdf2a9783512a35da4dc7676a66404b6f3826a8af9aad038722da6823R48
- https://github.com/rapidsai/rmm/pull/941/files#diff-4d2a8eefdf2a9783512a35da4dc7676a66404b6f3826a8af9aad038722da6823R99

This change only applies for non Project Flash builds: https://github.com/rapidsai/rmm/pull/941/files#diff-285f239638b8b5c5499ef5fda2e3db69adda7045e81556b4f2df478e99ab6263R71

But `rmm` is using Project Flash builds, so we also need to enable tests build for it.
As we are using the conda artifacts to run the tests in a Project Flash build, we need to add the `test` target when we are doing the `conda build`.

@harrism WDYT?

**Comment by harrism** [comment]
Ah, I see now. Unfortunately I'm out sick so will have to wait until next week unless you want to commit a fix.

**Comment by Ethyling** [comment]
@harrism The tests and benchmarks are now building on PRs, but we only run the tests currently. Should we also run the benchmarks?

**Comment by harrism** [comment]
No, don't run the benchmarks. Benchmarking is a separate effort and would likely not be done on every PR. We just need to make sure PRs don't break the benchmark build.

**Comment by harrism** [comment]
Thanks for the help, @Ethyling !

**Comment by harrism** [comment]
Is this ready to merge now? Still needs ops approval.

**Review by Ethyling — APPROVED** [review]

**Comment by Ethyling** [comment]
> Is this ready to merge now? Still needs ops approval.

We are good to merge

**Comment by harrism** [comment]
@gpucibot merge

**Pull request merged**